### ISSUE TITLE
bug: Error while compiling resources on Android 6.0

### ISSUE THREAD

**Issue by kitadai31** [kitadai31/revanced-manager-android6-7#5]
On Android 6.0, ReVanced Manager says "Aborting..." while compiling resources.

It seems like aapt2 issue.  
I have no idea how to fix it.

This is collection of screenshots and logs for researching this issue.

---

https://github.com/kitadai31/revanced-patches-android6-7/issues/1

![339745486_234174775832821_7794793368304987657_n](https://user-images.githubusercontent.com/111590878/232193515-5840c6ed-e2d3-44c9-a12f-6aea5c74770b.jpg)

![339547509_1398071697678185_1676525983522664396_n](https://user-images.githubusercontent.com/111590878/232193527-e3c807d0-85e3-4e8d-9017-cdae9fcf9884.jpg)

---

https://egg.5ch.net/test/read.cgi/android/1678959841/809

![](https://i.imgur.com/ZwZMoGy.jpg)

---

[**Magnimate**](https://github.com/Magnimate) April 18, 2023 from discussion

![Screenshot_20230418-124914](https://user-images.githubusercontent.com/108208865/232658105-900c2f62-170d-4856-8241-3733b939f1a7.png)

---
https://github.com/kitadai31/revanced-patches-android6-7/discussions/9

```
Error:
brut.androlib.AndrolibException: brut.common.BrutException: could not exec (exit code = 1): [/data/app/app.revanced.manager.flutter-1/lib/arm/libaapt2.so, compile, --dir, /data/user/0/app.revanced.manager.flutter/cache/patcher/tmp-NHKUJJ/cache/res, --legacy, -o, /data/user/0/app.revanced.manager.flutter/cache/patcher/tmp-NHKUJJ/cache/build/resources.zip]
at brut.androlib.res.AndrolibResources.aapt2Package(AndrolibResources.java:365)
at brut.androlib.res.AndrolibResources.aaptPackage(AndrolibResources.java:619)
at app.revanced.patcher.Patcher.save(Patcher.kt:126)
at app.revanced.manager.flutter.MainActivity.runPatcher$lambda-20(MainActivity.kt:212)
at app.revanced.manager.flutter.MainActivity.$r8$lambda$kxvfcFiIKARIH6oxYjk9gWWxjp0(MainActivity.kt)
at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda0.run(D8$$SyntheticClass)
at java.lang.Thread.run(Thread.java:818)
Caused by: brut.common.BrutException: could not exec (exit code = 1): [/data/app/app.revanced.manager.flutter-1/lib/arm/libaapt2.so, compile, --dir, /data/user/0/app.revanced.manager.flutter/cache/patcher/tmp-NHKUJJ/cache/res, --legacy, -o, /data/user/0/app.revanced.manager.flutter/cache/patcher/tmp-NHKUJJ/cache/build/resources.zip]
at brut.util.OS.exec(OS.java:99)
at brut.androlib.res.AndrolibResources.aapt2Package(AndrolibResources.java:361)
... 6 more
```

---
https://github.com/kitadai31/revanced-patches-android6-7/issues/13

```
Repacking patched apk
Compiling resources
An error occurred! Aborting
Error:
brut.androlib.AndrolibException: brut.common.BrutException: could not exec (exit code = 1): [/data/app/app.revanced.manager.flutter-1/lib/arm64/libaapt2.so, compile, --dir, /data/user/0/app.revanced.manager.flutter/cache/patcher/tmp-BXUFNS/cache/res, --legacy, -o, /data/user/0/app.revanced.manager.flutter/cache/patcher/tmp-BXUFNS/cache/build/resources.zip]
	at brut.androlib.res.AndrolibResources.aapt2Package(AndrolibResources.java:365)
	at brut.androlib.res.AndrolibResources.aaptPackage(AndrolibResources.java:619)
	at app.revanced.patcher.Patcher.save(Patcher.kt:126)
	at app.revanced.manager.flutter.MainActivity.runPatcher$lambda-20(MainActivity.kt:212)
	at app.revanced.manager.flutter.MainActivity.$r8$lambda$kxvfcFiIKARIH6oxYjk9gWWxjp0(MainActivity.kt)
	at app.revanced.manager.flutter.MainActivity$$ExternalSyntheticLambda0.run(D8$$SyntheticClass)
	at java.lang.Thread.run(Thread.java:818)
Caused by: brut.common.BrutException: could not exec (exit code = 1): [/data/app/app.revanced.manager.flutter-1/lib/arm64/libaapt2.so, compile, --dir, /data/user/0/app.revanced.manager.flutter/cache/patcher/tmp-BXUFNS/cache/res, --legacy, -o, /data/user/0/app.revanced.manager.flutter/cache/patcher/tmp-BXUFNS/cache/build/resources.zip]
	at brut.util.OS.exec(OS.java:99)
	at brut.androlib.res.AndrolibResources.aapt2Package(AndrolibResources.java:361)
	... 6 more
```

**Comment by kitadai31** [comment]
@tinyx3k That issue is unrelated to ReVanced Manager.
Please open new issue in https://github.com/kitadai31/revanced-patches-android6-7/issues/new/choose

**Comment by Jarusz** [comment]
The same issues happens on revanced builder on pc

**Comment by kitadai31** [comment]
It had been caused by some icon files in `overlay-buttons` patch, and it was fixed in [ravanced-patches v2.160.6](https://github.com/kitadai31/revanced-patches-android6-7/releases/tag/v2.160.6)
